src/gui/clips/SampleClipView.cpp

```
@@ -376,5 +385,27 @@ bool SampleClipView::splitClip( const TimePos pos )
 	else { return false; }
 }
 
+void SampleClipView::exportSampleBuffer()
+{
+	auto openFileDialog = FileDialog(nullptr, tr("Export audio file"), QString(), tr("Audio files (*.wav *.flac *.ogg *.mp3);;WAV (*.wav);;FLAC (*.flac);;OGG (*.ogg);;MP3 (*.mp3)"));
+
+
+	if (openFileDialog.exec() == QDialog::Accepted)
+	{
+		QStringList curSelectedFiles(openFileDialog.selectedFiles());
+		if (curSelectedFiles.isEmpty() == false)
+		{
+			QString outputFileName = curSelectedFiles.first();
```

It seems that you silently drop all other files if several files are selected in the dialog. Please instead disallow the selection of multiple files in the first place.

---

How to fix this?

---

I took the liberty to push some changes with commit 3566660591f to save some back and forth. The gist of it is that `FileDialog::getSaveFileName` is used to retrieve the file name. This gives us lots of functionality automatically, e.g. the request if an existing file should be overwritten. The previous implementation did not do this which I had to learn the hard way. :sweat_smile:

It also removes the appending of the `.flac` file extension and lets the users decide the name.